src/github.com/couchbase/sync_gateway/db/change_cache.go

```diff
@@ -282,13 +418,29 @@ func (c *changeCache) _addToCache(change *LogEntry) base.Set {
 	addedTo := make([]string, 0, 4)
 	ch := change.Channels
 	change.Channels = nil // not needed anymore, so free some memory
+
+	// If it's a late sequence, we want to add to all channel late queues within a single write lock,
+	// to avoid a changes feed seeing the same late sequence in different iteration loops (and sending
+	// twice)
+	if change.Skipped {
+		c.LateSeqLock.Lock()
+	}
+
```

Comment:
Rather than having the Lock() here and the Unlock() lower in the file, I think this should be wrapped in a function where the Lock() and defer Unlock() happen next to each other. 

See https://github.com/couchbase/sync_gateway/commit/d80857e143d5c77ee2c7dbe8b1a1a89b488a94be